src/layers/domain/core/device/tests/test_device_v2.py

```diff
@@ -209,9 +213,66 @@ def test_device_add_tag(device_v2: Device):
     with pytest.raises(DuplicateError):
         device_v2.add_tag(bar="second", foo="first")
 
-    assert [tag.value for tag in device_v2.tags] == [
-        "<<bar##second>>##<<foo##first>>",
-        "<<bar##second>>##<<baz##third>>##<<foo##first>>",
-    ]
+    assert sorted(tag.value for tag in device_v2.tags) == sorted(
+        [
+            "<<bar##second>>##<<foo##first>>",
+            "<<bar##second>>##<<baz##third>>##<<foo##first>>",
+        ]
+    )
 
     assert event_2.updated_on > event_1.updated_on
+
+    event_3 = device_v2.clear_tags()
+    assert isinstance(event_3, DeviceTagsClearedEvent)
+    assert event_3.updated_on > event_2.updated_on
+    assert device_v2.tags == set()
+
+
+def test_device_add_tags_in_one_go(device_v2: Device):
+    event = device_v2.add_tags(
+        tags=[
+            dict(foo="first", bar="second"),
+            dict(foo="first", bar="second", baz="third"),
+        ]
+    )
+    assert isinstance(event, DeviceTagsAddedEvent)
+    assert event.updated_on is not None
+    assert event.updated_on == device_v2.updated_on
+
+    with pytest.raises(DuplicateError):
+        device_v2.add_tags([dict(bar="second", foo="first")])
+
+    assert sorted(tag.value for tag in device_v2.tags) == sorted(
+        [
+            "<<bar##second>>##<<foo##first>>",
+            "<<bar##second>>##<<baz##third>>##<<foo##first>>",
+        ]
+    )
+
+
+def test_device_tag():
+    tag = DeviceTag(foo="bar", boo="far")
+    tag_as_dict = tag.dict()
+    reconstituted_tag = DeviceTag(__root__=tag.__root__)
+
+    assert tag_as_dict == tag.__root__
+    assert reconstituted_tag == tag
+    assert reconstituted_tag in {tag}
+
+
+def test_device_tag():
```

Comment:
same test name

Reply:
I think this has been duplicated somewhere along the way.. maybe when the `__root__` change was made. Should only need the first one now? Unless I'm getting confused and they're still 2 separate tests with the same name..